.github/workflows/check-devcontainer.yml

```diff
@@ -3,17 +3,51 @@ on:
   pull_request:
     paths:
       - .devcontainer/**
+      - pyproject.toml
+      - poetry.lock
   push:
     branches:
       - master
     paths:
       - .devcontainer/**
 jobs:
   Build-Devcontainer:
+    permissions:
+      contents: write
     runs-on: ubuntu-latest
     steps:
     - name: Checkout
       uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - name: install poetry
```

Comment:
I remember github actions for installing poetry, do they have advantages over manually installing?